central/compliance/datastore/datastore_impl.go

```diff
@@ -140,18 +135,6 @@ func (ds *datastoreImpl) StoreRunResults(ctx context.Context, results *storage.C
 		return sac.ErrResourceAccessDenied
 	}
 
-	ds.storedAggregationMutex.Lock()
-	defer ds.storedAggregationMutex.Unlock()
-
-	defer func() {
-		// Atomic because it will be atomically read outside the mutex
-		atomic.AddUint64(&ds.aggregationSequenceNumber, 1)
-	}()
-
-	if err := ds.storage.ClearAggregationResults(ctx); err != nil {
-		log.Errorf("unable to clear old stored aggregations: %v", err)
-	}
-
 	return ds.storage.StoreRunResults(ctx, results)
```

Comment:
The datastore level `StoreRunResults` is called from a goroutine.
After the removal of `storedAggregationMutex`, the call to the storage level `StoreRunResults` is not protected anymore against concurrent access. It might be worth checking that the storage-level function does not do any concurrent access.

Reply:
Looking closer, there is nothing in the storage object that would require protection against concurrent access (apart from the connection to send requests, but this should be handled at DB client library level). Successive calls are made to the various `Compliance` entity stores to upsert elements of the run results.